reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerConfig.java

```diff
@@ -344,6 +346,7 @@ public Function<String, String> uriTagValue() {
 		this.httpMessageLogFactory = ReactorNettyHttpMessageLogFactory.INSTANCE;
 		this.maxKeepAliveRequests = -1;
 		this.minCompressionSize = -1;
+		this.compressionOptions = HttpCompressionOptionsSpec.provideDefault();
```

Comment:
Can we not initialise this with a default, or maybe it can be empty? The default will be used only when the compression is enabled. In Reactor Netty we try to not initialise things that will not be used.

Reply:
@violetagg 
Thank you for the suggestion!
I’ve updated the implementation in commit c28d954 to set compressionOptions to null as the default state when compression is disabled.